test/e2e/osa_test.go

```diff
@@ -30,10 +34,52 @@ var _ = BeforeSuite(func() {
 	c.cs, err = managedcluster.ReadConfig("../../_data/containerservice.yaml")
 	Expect(err).NotTo(HaveOccurred())
 
-	c.kc, err = managedcluster.ClientsetFromConfig(c.cs)
+	// Default to system:admin for all tests. Use kubeClient() selectively
+	// to test with different permissions.
+	c.kc, err = managedcluster.ClientsetFromConfig(c.cs.Config.AdminKubeconfig)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Create a namespace for running tests and assign osadmin
+	// as a project admin to it. Delegate permissions to osadmin
+	// to allow us to run tests in a non-privileged context.
+	_, err = c.kc.CoreV1().Namespaces().Create(&v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "e2e-tests",
```

Comment:
are we using one namespace for all tests? if we introduce parallelism to the tests will it not clash?

Reply:
I have seen this now in origin conformance testing. If in example few tests use same namespace and one of them is testing "network policy" or rbac restriction it automatically can flake another test in the same namespace, despite the fact it uses different resources.

Reply:
> can flake another test in the same namespace, despite the fact it uses different resources.

Can you elaborate? Is it namespace cleanup that triggers the flake or something else?

Reply:
that could be too. I think each test case should create unique namespace for itself and clean after.

Reply:
I am against namespace cleanup. It's probably one of the flakiest things I have ever seen in origin e2e.